astropy/convolution/convolve.py

```diff
@@ -1033,7 +1076,20 @@ def convolve_models(model, kernel, mode="convolve_fft", **kwargs):
     else:
         raise ValueError(f"Mode {mode} is not supported.")
 
-    return CompoundModel(operator, model, kernel)
+    if bounding_box is None or resolution is None:
+        none_kws = " and ".join(
+            [
+                k
+                for k, v in {
+                    "bounding_box": bounding_box,
+                    "resolution": resolution,
+                }.items()
+                if v is None
+            ]
+        )
```

Comment:
In my opinion this list comprehension cannot be clever *and* easy to read, and I'd favour readability, so I suggest making it dumb :)
```suggestion
        none_kws_list = []
        if bounding_box is None:
            none_kws_list.append("bounding_box")
        if resolution is None:
            none_kws_list.append("resolution")
        none_kws = " and ".join(none_kws_list)
```